bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -40,25 +41,42 @@
 	}
 
 	/**
-	 * Publish the artifact. This includes making any events that may be derived
+	 * Posts the artifact. This includes making any events that may be derived
 	 * from it, and broadcasting a notification that the artifact is ready for
 	 * further analysis.
 	 *
-	 * @param artifact The artifact to be published.
+	 * @param artifact The artifact to be posted.
 	 *
-	 * @throws BlackboardException If there is a problem publishing the
-	 *                             artifact.
+	 * @throws BlackboardException If the blackboard is closed.
+	 * @throws TskCoreException    If there is a problem posting the artifact.
 	 */
-	public synchronized void postArtifact(BlackboardArtifact artifact) throws BlackboardException {
+	public synchronized void postArtifact(BlackboardArtifact artifact) throws BlackboardException, TskCoreException {
```

Comment:
for me, tskcore has always been about problems accessing the DB.  The blackboard exception is specifically because the blackboard is closed, and is sort of inherited from the Autopsy version of this class.  I have no strong feeling though.

Reply:
TskCoreException == database issue is pretty reasonable, to be sure. However, sticking to BlackboardException alone does give the user the possibility of distinguishing between SleuthkitCase (etc.) method call exceptions and Blackboard method call exceptions (but of course, that can also be done with multiple try blocks). Also, Blackboard potentially abstracts away the sort of unwritten TskCoreException == database issue association, which, to be exacting, is not always true. Giving it some more thought, I am still in favor of distinguished BlackboardExceptions. The clincher for me is that strategy allows for adding more debug info into the exception messages, potentially augmenting what we get from the "lower level" TskCoreExceptions.

Reply:
got it